### PR TITLE
Reorganized docs to add Bare Metal

### PR DESCRIPTION
**Pull request by chrisnegus**
*Description of changes:*

This PR modifies the Reference -> Config section of EKS-A docs to:

* Have “vSphere Configuration”, “Bare Metal”, and new “Optional” section under Config.
* Move existing configuration sections (CNI, etcd, IAM for pods, IAM Auth, OIDC, GitOps, Proxy and registry) and new Multus section to the Optional section.
* Add links to vSphere and Bare Metal config docs to Optional sections that are supported by those provider types.

This PR also creates placeholder files for new Bare Metal Reference section (docs/reference/baremetal), including:
* _index
* Hardware requirements
* Preparation
* iPXE and Tinkerbell
* Custom OVAs
* Custom HookOS.

Also, updated links to the files that moved.

Questions:
* Should Optional sections be in any particular order?
* Within each Optional section, should we identify whether it is supported by vSphere and/or Bare Metal or is simply listing supported optional section in the main vSphere and Bare Metal docs enough?

### CODE REVIEW

**Comment by chrisnegus** [comment]
This is what the reorganized Config section looks like from this PR:
configref

**Comment by chrisnegus** [comment]
Here's the new Bare Metal reference section:
baremetal

File: docs/content/en/docs/reference/baremetal/bare-custom-hookos.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+title: "Customize HookOS for EKS Anywhere on Bare Metal"
+linkTitle: "Customize HookOS for EKS Anywhere on Bare Metal"
```
Comment (rothgar) [inline docs/content/en/docs/reference/baremetal/bare-custom-hookos.md:3]: Can we remove all the "for EKS Anywhere on Bare Metal" from the link titles? It makes the side bar harder to read and a bit cluttered/redundant

Comment (chrisnegus) [reply docs/content/en/docs/reference/baremetal/bare-custom-hookos.md]: Yes. I agree. Done.

**Pull request merged**